enhancements/installer/aws-cross-account-dns-zone.md

```diff
@@ -0,0 +1,294 @@
+	// +optional
+	AWS *AWSDNSSpec `json:"aws,omitempty"`
+}
+
+// AWSDNSSpec contains DNS configuration specific to the Amazon Web Services cloud provider.
+type AWSDNSSpec struct {
+	// privateZoneIAMRole contains the ARN of a role that should be assumed when performing
+	// operations on the cluster's private hosted zone specified in the cluster DNS config.
+	// When left empty, no role should be assumed.
+	// +kubebuilder:validation:Pattern:=`^arn:(aws|aws-cn|aws-us-gov):iam:[a-z0-9-]+:[0-9]{12}:role\/.*$`
+	// +optional
+	PrivateZoneIAMRole string `json:"privateZoneIAMRole,omitempty"`
+}
+
+```
+
+The name of the field is discussed further in open questions.
+
+#### Install Config
+
+`platform.aws.hostedZoneRole` would be added to complement the existing `platform.aws.hostedZone` field.
+
+An example of `platform` from `install-config.yaml`, specifying subnets of the shared VPC,
+a hosted zone ID, and a role to assume:
+
+```yaml
+platform:
+  aws:
+    region: us-east-2
+    subnets:
+    - subnet-0ef71407239a73558
+    - subnet-00e576c91f8578661
+    - subnet-041af6e1bbf29c2ef
+    - subnet-037ab7dc0cd396721
+    - subnet-0adda52a1007fc74f
+    - subnet-09bd88678851acab6
+    hostedZone: Z0743278N5ELJPLYA3H6
+    hostedZoneRole: arn:aws:iam::<account-a>:role/<role-name>
+```
+Please see Open Questions for further discussion of the install config.
+
+
+
+### Implementation Details/Notes/Constraints [optional]
+
+ With the implementation of this API, the installer and cluster components should follow a
+ relatively straightforward pattern: if the role field is not empty,
+ call [`AssumeRole`](https://docs.aws.amazon.com/sdk-for-go/api/aws/credentials/stscreds/)
+ when performing operations on the private hosted zone.
+
+### Risks and Mitigations
+
+n/a
+
+### Drawbacks
+
+Up to this point, DNS config has been platform agnostic. There are no obvious drawbacks to introducing
+platform dependencies in this config, but it should be carefully considered.
+## Design Details
+
+### Open Questions [optional]
+
+1. The suggested field name for the API is clunky, but has the advantage of being least ambiguous.
+Some other possible candidates would be:
+- sharedVPCRole
+- privateDNSRole
+- privateZoneRole
+- privateDNSZoneRole
+- dnsRole
+- dnsZoneRole
+- privateHostedZoneRole
+2. In the install config, should we consider deprecating `aws.platform.hostedZone` in order to
+create a struct that can nest both the `id` and `role` fields? Using the same example as above,
+the install config would look like:
+
+```yaml
+platform:
+  aws:
+    region: us-east-2
+    subnets:
+    - subnet-0ef71407239a73558
+    - subnet-00e576c91f8578661
+    - subnet-041af6e1bbf29c2ef
+    - subnet-037ab7dc0cd396721
+    - subnet-0adda52a1007fc74f
+    - subnet-09bd88678851acab6
+    hostedZone: #deprecated
+    privateHostedZone:
+      id: Z0743278N5ELJPLYA3H6
+      role: arn:aws:iam::<account-a>:role/<role-name>
```

Comment:
> is there any claim for having the Public Hosted Zone (baseDomain) in another account?

I don't believe so, but I am also looking for a definitive answer. The reason I don't believe this is the case, is because with a public zone, if the authoritative dns zone belongs to a different account, users should already be able to delegate to another public DNS zone in the cluster account.